### PR TITLE
Implement Fake new registry pattern

### PR DESCRIPTION
**Pull request by diegomarangoni**
EES-4625
This PR implement new registry pattern into Fake backend

### CODE REVIEW

File: pkg/backends/fake/fake.go [hunk 1]
```diff
@@ -5,28 +5,35 @@ import (
...
+	imageName = "alpine"
+	imageTag  = "3.12.0"
 )
 
 // Fake enables the controller to run a LoadTest using Fake load provider which simulates load test
 type Fake struct {
+	backend    *Backend
 	kubeClient kubernetes.Interface
 	loadTest   *loadTestV1.LoadTest
 	logger     *zap.Logger
```
Comment (boekkooi-fresh) [inline pkg/backends/fake/fake.go:23]: Do you still need the Logger and KubeClient in here?

Comment (diegomarangoni) [reply pkg/backends/fake/fake.go]: nope, removed

File: pkg/backends/fake/backend.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
+package fake
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	batchV1 "k8s.io/api/batch/v1"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
```
Comment (vgarvardt) [inline pkg/backends/fake/backend.go:10]: k8sAPIErrors pls

File: pkg/backends/fake/backend.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	loadTestStatus.Phase = getLoadTestPhaseFromJob(job.Status)
+	loadTestStatus.JobStatus = job.Status
+
+	return nil
+}
+
+// newMasterJob creates a new job which runs the Fake master pod
+func (b *Backend) newMasterJob(loadTest loadTestV1.LoadTest) *batchV1.Job {
+	imageRef := fmt.Sprintf("%s:%s", loadTest.Spec.MasterConfig.Image, loadTest.Spec.MasterConfig.Tag)
+	if "" == loadTest.Spec.MasterConfig.Image || "" == loadTest.Spec.MasterConfig.Tag {
```
Comment (vgarvardt) [inline pkg/backends/fake/backend.go:122]: this is C-style =) this type of error (missing of one `=`) is not possible in golang `if` =)

**Comment by vgarvardt** [comment]
👍

**Pull request merged**